### PR TITLE
Use the daemon replug for Thunderbolt

### PR DESCRIPTION
**Pull request by superm1**

### CODE REVIEW

**Comment by hughsie** [comment]
Looks logically correct, but CI failure: `remove_leftover_devices’ defined but not used` -- once CI passes LGTM. Thanks.

**Comment by superm1** [comment]
It's actually a bigger failure than the CI failure shows, I need to rework some of the test suite too.  I'll follow it back up.

**Comment by superm1** [comment]
(Don't expect that current one ^ to work still)

**Comment by superm1** [comment]
@gicmo might you be able to help with the changes to the test suite?  I am running into problems where the file changed callback doesn't fire, but I'm a bit confused why.

**Comment by gicmo** [comment]
Yeah, I can have a look!

**Comment by superm1** [comment]
> Yeah, I can have a look!

Thanks!  Feel free to push right to this branch if you figure it out.

**Comment by gicmo** [comment]
Got interrupted, but had a quick look, nothing obvious; indeed it seems we don't get/catch the udev event anymore. I need to head out now, but will look again later/tomorrow.

There is some code that can be removed, that is actually involved in udev handling, but removing it did not change anything for me:
```diff
diff --git a/plugins/thunderbolt/fu-plugin-thunderbolt.c b/plugins/thunderbolt/fu-plugin-thunderbolt.c
index abf0a061..5200b40f 100644
--- a/plugins/thunderbolt/fu-plugin-thunderbolt.c
+++ b/plugins/thunderbolt/fu-plugin-thunderbolt.c
@@ -36,10 +36,6 @@ typedef void (*UEventNotify) (FuPlugin	  *plugin,
 struct FuPluginData {
 	GUdevClient   *udev;
 
-	/* in the case we are updating */
-	UEventNotify   update_notify;
-	gpointer       update_data;
-
 	/* the timeout we wait for the device
 	 * to be updated to re-appear, in ms.
 	 * defaults to:
@@ -410,20 +406,12 @@ udev_uevent_cb (GUdevClient *udev,
 		gpointer     user_data)
 {
 	FuPlugin *plugin = (FuPlugin *) user_data;
-	FuPluginData *data = fu_plugin_get_data (plugin);
 
 	if (action == NULL)
 		return TRUE;
 
 	g_debug ("uevent for %s: %s", g_udev_device_get_sysfs_path (device), action);
 
-	if (data->update_notify != NULL) {
-		g_debug ("using update notify handler for uevent");
-
-		data->update_notify (plugin, device, action, data->update_data);
-		return TRUE;
-	}
-
 	if (g_str_equal (action, "add")) {
 		fu_plugin_thunderbolt_add (plugin, device);
 	} else if (g_str_equal (action, "remove")) {
-- 
2.17.1
```

**Comment by gicmo** [comment]
There is some kind of corruption going on, because executing the test in isolation works (`LD_PRELOAD=libumockdev-preload.so.0 G_MESSAGES_DEBUG=all build/plugins/thunderbolt/thunderbolt-self-test -p "/thunderbolt/update{failing}`). So something is messing with the `root_one` memory.  Really heading out now.

**Comment by superm1** [comment]
It seems like what's happening is there isn't a mainloop running (since it was torn out), so the callbacks aren't working correctly.  Since we actually get a return between steps now I'm tempted to just remove this callback code in the first place.

**Comment by superm1** [comment]
The problem causing the overwriting memory was actually also due to the lack of a mainloop running.  Events got queued up from the "last" test run and when the mainloop for the next test run started, they fired at that time.  This explained the garbage in those tests.

So what I decided to do was call sync between the various commands that devices should come and go (after `update`, eg after `update_attach`) and this seems to clean up any pending events.

This PR was more surgery than I expected, but I think `-300` and using the daemon now is a net win for a 1000 line plugin.

File: plugins/thunderbolt/fu-plugin-thunderbolt.c [hunk 1]
```diff
@@ -101,9 +91,8 @@ fu_plugin_thunderbolt_udev_get_id (GUdevDevice *device,
 
 	guint64 id;
```
Comment (hughsie) [inline plugins/thunderbolt/fu-plugin-thunderbolt.c]: I think you need `id = 0` here or coverity is going to complain.

**Comment by hughsie** [comment]
Much simpler! LGTM, one tiny nit. Thanks!

**Pull request merged**